scrapy/extensions/feedexport.py

```diff
@@ -285,11 +276,49 @@ def close_spider(self, spider):
             deferred_list.append(d)
         return defer.DeferredList(deferred_list) if deferred_list else None
 
+    def _start_new_batch(self, previous_batch_slot, uri, feed, spider):
+        """
+        Redirect the output data stream to a new file.
+        Execute multiple times if 'FEED_STORAGE_BATCH' setting is greater than zero.
+        """
+        if previous_batch_slot is not None:
+            previous_batch_slot.exporter.finish_exporting()
+            previous_batch_slot.storage.store(previous_batch_slot.file)
+        storage = self._get_storage(uri)
+        file = storage.open(spider)
+        exporter = self._get_exporter(
+            file=file,
+            format=feed['format'],
+            fields_to_export=feed['fields'],
+            encoding=feed['encoding'],
+            indent=feed['indent']
+        )
+        slot = _FeedSlot(file, exporter, storage, uri, feed['format'], feed['store_empty'])
+        if slot.store_empty:
+            slot.start_exporting()
+        return slot
+
+    def _get_uri_of_partial(self, template_uri, feed, spider):
```

Comment:
I guess you shouldn't force the timestamp in the file name.
Just log a warning if `%(time)s` is not in the URI as it will cause errors.
Maybe, we could add a new parameter, like `%(batch_id)s` also

Reply:
If I understand correctly, the user must specify %(time)s and %(batch_id)s in the path and the input should be
scrapy crawl quotes -o res/test%(time)s.json - something like that?
Is %(batch_id)s a serial number of current batch?

Reply:
Yup, the users should set the parameter as they wish.
We should warn them if that parameter is missing.

The new `batch_id` parameter can be just a sequential id starting at 1..